### PR TITLE
Feature/moa support

### PR DESCRIPTION
**Pull request by creisle**
New Features
- MOAlmonac loader
- Add MOA evidence levels and vocab to data files

Improvements
- Replace request-promise library with node-fetch (v2) library since the former is not longer maintained
- Update some dependencies that had severe audit vulnerabilities
- Make default source in evidence levels graphkb instead of vicc
- get source definition defaults from source.js to avoid having to add them to data files

### CODE REVIEW

File: src/moa/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# MOAlmonac
+
+Load variant assestions from the [Molecular Oncology Almonac](https://moalmanac.org)[^1].
```
Comment (elewis2) [inline src/moa/README.md:3]: sp - Almonac -> Almanac

Comment (creisle) [reply src/moa/README.md:3]: wow, i did that in a lot of places and didn't even notice :O, will fix

**Review by elewis2 — CHANGES_REQUESTED** [review]
just the one typo - Almonac -> Almanac

File: README.md [hunk 2]
```diff
@@ -58,6 +58,7 @@ docker run bcgsc/pori-graphkb-loader --help
 - [COSMIC](./src/cosmic)
 - [DGIdb](./src/dgidb)
 - [DoCM](./src/docm)
+- [MOAlmonac](./src/moa)
```
Comment (elewis2) [inline README.md]: also here

Comment (creisle) [reply README.md]: I think you might be looking at an old view that doesn't include the fix commit? On mine it is showing up as changed already

File: src/moa/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+We have linked the `predictive_implication` from MOA to specific evidence levels in GraphKB
+
+| MOA               | AMP               | CIViC | IPR |
+| ----------------- | ----------------- | ----- | --- |
+| Clinical evidence | Level B (Tier II) | B     | B   |
+| Clinical trial    | Level B (Tier II) | B     | B   |
+| FDA-Approved      | Level A (Tier I)  | A     | A   |
+| Guideline         | Level A (Tier I)  | A     | A   |
+| Inferential       |                   | E     | E   |
+
```
Comment (cgrisdale) [inline src/moa/README.md:39]: Should we add: MOA-preclinical -> Level D (Tier II) -> D -> D

Comment (creisle) [reply src/moa/README.md:40]: Yup! I just missed that one. Added

File: src/moa/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+### Category Variant Mapping
+
+Specific MOA variant types are mapped as category variants in GraphKB:
+
+| MOA feature_type         | MOA field               | MOA field value | GraphKB reference1 class | GraphKB reference1         | GraphKB type      |
+| ------------------------ | ----------------------- | --------------- | ------------------------ | -------------------------- | ----------------- |
+| microsatellite_stability | status                  | MSI-High        | Signature                | microsatellite instability | high signature    |
+| mutational_signature     | cosmic_signature_number | #               | Signature                | SBS#                       | signature present |
+| knockdown                | n/a                     | `<gene name>`   | Feature                  | `<gene name>`              | knockdown         |
+
```
Comment (cgrisdale) [inline src/moa/README.md:50]: Do we want an example for Mutation Burden as well? It has slightly different 'attributes' than MSI.
https://moalmanac.org/api/assertions/808

Comment (creisle) [reply src/moa/README.md:50]: I left it for now b/c I wasn't totally sure what to do with it https://github.com/bcgsc/pori_graphkb_loader/issues/38#issuecomment-1020561403

we could add it but without high/low it won't really be useful matching-wise anyway

File: src/moa/README.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+Load variant assestions from the [Molecular Oncology Almonac](https://moalmanac.org)[^1].
+
+```bash
+node bin/load.js api moa
+```
+
+## Loading Assumptions
+
+- A specific drug ontology is not used/given and therefore drugs are matched by name
+- A specific gene ontology is not specified and therefore we default to using Entrez genes as they are popular amongst other knowlege bases
```
Comment (cgrisdale) [inline src/moa/README.md:12]: Typo "knowlege" -> "knowledge"

**Review by elewis2 — APPROVED** [review]

**Review by cgrisdale — APPROVED** [review]

File: src/moa/README.md (hunk 5 above)
Comment (creisle) [reply src/moa/README.md:12]: fixed

**Pull request merged**